hub/graphql/types.py

```diff
@@ -9,28 +9,74 @@
 
 from hub import models
 
+from .utils import key_resolver
 
-@strawberry_django.type(models.Area)
-class Area:
+
+@strawberry_django.filters.filter(
+    procrastinate.contrib.django.models.ProcrastinateJob, lookups=True
+)
+class QueueFilter:
     id: auto
-    mapit_id: auto
-    gss: auto
-    name: auto
-    area_type: "AreaType"
-    geometry: auto
-    overlaps: List["Area"]
+    status: auto
+    queue_name: auto
+    task_name: auto
+    scheduled_at: auto
+    attempts: auto
+    external_data_source_id: Optional[str]
+
+    def filter_external_data_source_id(self, queryset, info, value):
+        return queryset.filter(args__external_data_source_id=value)
 
 
-@strawberry_django.type(models.AreaType)
-class AreaType:
+@strawberry_django.type(
+    procrastinate.contrib.django.models.ProcrastinateJob,
+    filters=QueueFilter,
+    pagination=True,
+)
+class QueueJob:
     id: auto
-    name: auto
-    area_type: auto
-    description: auto
+    queue_name: auto
+    task_name: auto
+    lock: auto
+    args: auto
+    status: auto
+    scheduled_at: auto
+    attempts: auto
+    queueing_lock: auto
+    events: List["QueueEvent"]
 
     @strawberry_django.field
-    def areas(self) -> List[Area]:
-        return self.area_set.all()
+    def last_event_at(self, info) -> datetime:
+        return (
+            procrastinate.contrib.django.models.ProcrastinateEvent.objects.filter(
+                job_id=self.id
+            )
+            .order_by("-at")
+            .first()
+            .at
+        )
+
+    @classmethod
+    def get_queryset(cls, queryset, info, **kwargs):
```

Comment:
I think this is good enough for now, but one would want to have some robust tests around this in future, as this check has a lot riding on it. i.e. it gate people from one organisation having access to data from another. On the upside, this makes sharing logics a lot easier potentially.